pipeline/runtime/common.go

```diff
@@ -83,3 +86,23 @@ func fetchOutputVariables(outputFile string, out io.Writer) (map[string]string,
 	}
 	return outputs, nil
 }
+
+// setTiEnvVariables sets the environment variables required for TI
+func setTiEnvVariables(step *spec.Step, config *api.TIConfig) {
+
+	if config == nil {
+		return
+	}
+	if step.Envs == nil {
+		step.Envs = map[string]string{}
+	}
+
+	envMap := step.Envs
+	envMap[ti.TiSvcEp] = config.URL
+	envMap[ti.TiSvcToken] = config.Token
```

Comment:
I suggest encoding this instead of exporting directly. Even a basic encoding scheme is fine.

Reply:
Ack.